src/components/medialibrarycreator/medialibrarycreator.js

```diff
@@ -2,30 +2,48 @@ define(["loading", "dialogHelper", "dom", "jQuery", "components/libraryoptionsed
     "use strict";
 
     function onSubmit(e) {
-        if (e.preventDefault(), e.stopPropagation(), 0 == pathInfos.length) return require(["alert"], function(alert) {
-            alert({
-                text: Globalize.translate("PleaseAddAtLeastOneFolder"),
-                type: "error"
-            })
-        }), !1;
-        var form = this,
-            dlg = $(form).parents(".dialog")[0],
-            name = $("#txtValue", form).val(),
-            type = $("#selectCollectionType", form).val();
-        "mixed" == type && (type = null);
-        var libraryOptions = libraryoptionseditor.getLibraryOptions(dlg.querySelector(".libraryOptions"));
-        return libraryOptions.PathInfos = pathInfos, ApiClient.addVirtualFolder(name, type, currentOptions.refresh, libraryOptions).then(function() {
-            hasChanges = !0, dialogHelper.close(dlg)
-        }, function() {
-            require(["toast"], function(toast) {
-                toast(Globalize.translate("ErrorAddingMediaPathToVirtualFolder"))
-            })
-        }), !1
+        e.preventDefault();
+        e.stopPropagation();
+
+        if (isCreating) return false;
+
+        if (pathInfos.length == 0) {
+            require(["alert"], function(alert) {
+                alert({
+                    text: Globalize.translate("PleaseAddAtLeastOneFolder"),
+                    type: "error"
+                })
+            });
+        } else {
+            isCreating = true;
+            loading.show();
+
+            var form = this,
+                dlg = $(form).parents(".dialog")[0],
+                name = $("#txtValue", form).val(),
+                type = $("#selectCollectionType", form).val();
+            if (type == "mixed") type = null;
+            var libraryOptions = libraryoptionseditor.getLibraryOptions(dlg.querySelector(".libraryOptions"));
+            libraryOptions.PathInfos = pathInfos;
+            ApiClient.addVirtualFolder(name, type, currentOptions.refresh, libraryOptions).then(function() {
+                hasChanges = true;
+                isCreating = false;
+                loading.hide();
+                dialogHelper.close(dlg);
+            }, function() {
+                require(["toast"], function(toast) {
+                    toast(Globalize.translate("ErrorAddingMediaPathToVirtualFolder"))
+                })
+                isCreating = false;
+                loading.hide();
+            });
+        }
+        return false;
     }
 
     function getCollectionTypeOptionsHtml(collectionTypeOptions) {
         return collectionTypeOptions.filter(function(i) {
-            return !1 !== i.isSelectable
+            return i.isSelectable !== false
```

Comment:
isn't it enough to do
```suggestion
            return i.isSelectable
```